src/errors.rs

```diff
@@ -31,8 +32,9 @@ impl std::error::Error for Error {
         match self {
             Error::Request(inner) => Some(inner),
             Error::Config(_) => None,
```

Comment:
Maybe worth just using wildcard match for all these `None` cases?
```suggestion
            _=> None,
```